### PR TITLE
PEP 0466 - SSL - possible fix to default to unverified context when n…

### PR DESCRIPTION
**Pull request by duffolonious**
…o cert information is specified (for python 2.7.9+)

### CODE REVIEW

File: pyVmomi/SoapAdapter.py [hunk 1]
```diff
@@ -1208,6 +1208,8 @@ def __init__(self, host='localhost', port=443, ns=None, path='/sdk',
       if cacertsFile:
          self.schemeArgs['ca_certs'] = cacertsFile
          self.schemeArgs['cert_reqs'] = ssl.CERT_REQUIRED
+      if self.schemeArgs == {}:
+         self.schemeArgs['context'] = ssl._create_unverified_context()
```
Comment (hartsock) [inline pyVmomi/SoapAdapter.py]: My only comment here is to figure out how to incorporate the other work on switches to allow the class to toggle between this unverified default behavior and the stronger verified behavior that the security folks want to see into this patch.

**Comment by hartsock** [comment]
We're seeing some build problems [here](https://travis-ci.org/vmware/pyvmomi/jobs/67236782#L718) and [here](https://travis-ci.org/vmware/pyvmomi/jobs/67236782)... but I think this is significant because you've identified the right place to catch where the other contexts were coming from. I wasn't finding this on my own. 

Possible courses of action:
* pull this into my other PR and modify it to pass the build and use the switches we are defining
* have @parthgala incorporate this into his work
* take this under advisement and use it for other PR

**Comment by duffolonious** [comment]
I'll take a look, probably need to test on an older version of python?  I'll try out something with older 2.7.

**Comment by hartsock** [comment]
Thanks @duffolonious I was trying to work out how to optionally toggle the unverified behavior [here](https://github.com/vmware/pyvmomi/pull/252/files). If I finish this work myself I'll try and cherry-pick your patch and fold it into that work somehow. However, @parthgala is heading up the project now and he's working on a solution as well. I'm waiting to see what he will do.

**Comment by duffolonious** [comment]
Updated to at least do nothing in other non-2.7 versions of python.

**Comment by duffolonious** [comment]
Anything to keep this from being pulled?

**Comment by hartsock** [comment]
I want to use this in conjunction with one of the other pull requests:
* the default behavior should disallow unverified contexts
* the unverified context should be an optional override

I'm hoping to steal some time to take your contribution and mix it with one of the other contributions. Once I have a few hours to spend on that I'll roll out a "beta" release or something to allow people to review the behaviors.

**Comment by flaub** [comment]
Any news on this?

**Comment by tianhao64** [comment]
This issue has been addressed by 92c1de5. Connect method now accept a sslContext parameter.

**Pull request closed without merging**